distributed/worker.py

```diff
@@ -1933,1365 +1805,142 @@ def _(**kwargs):
-        return cause
-
-    def _update_metrics_received_data(
-        self,
-        start: float,
-        stop: float,
-        data: dict[str, Any],
-        cause: TaskState,
-        worker: str,
-    ) -> None:
-
-        total_bytes = sum(self.tasks[key].get_nbytes() for key in data)
-
-        cause.startstops.append(
-            {
-                "action": "transfer",
-                "start": start + self.scheduler_delay,
-                "stop": stop + self.scheduler_delay,
-                "source": worker,
-            }
-        )
-        duration = (stop - start) or 0.010
-        bandwidth = total_bytes / duration
-        self.incoming_transfer_log.append(
-            {
-                "start": start + self.scheduler_delay,
-                "stop": stop + self.scheduler_delay,
-                "middle": (start + stop) / 2.0 + self.scheduler_delay,
-                "duration": duration,
-                "keys": {key: self.tasks[key].nbytes for key in data},
-                "total": total_bytes,
-                "bandwidth": bandwidth,
-                "who": worker,
-            }
-        )
-        if total_bytes > 1_000_000:
-            self.bandwidth = self.bandwidth * 0.95 + bandwidth * 0.05
-            bw, cnt = self.bandwidth_workers[worker]
-            self.bandwidth_workers[worker] = (bw + bandwidth, cnt + 1)
-
-            types = set(map(type, data.values()))
-            if len(types) == 1:
-                [typ] = types
-                bw, cnt = self.bandwidth_types[typ]
-                self.bandwidth_types[typ] = (bw + bandwidth, cnt + 1)
-
-        if self.digests is not None:
-            self.digests["transfer-bandwidth"].add(total_bytes / duration)
-            self.digests["transfer-duration"].add(duration)
-        self.counters["transfer-count"].add(len(data))
-        self.incoming_count += 1
-
-    @fail_hard
-    @log_errors
-    async def gather_dep(
-        self,
-        worker: str,
-        to_gather: Collection[str],
-        total_nbytes: int,
-        *,
-        stimulus_id: str,
-    ) -> StateMachineEvent | None:
-        """Gather dependencies for a task from a worker who has them
-
-        Parameters
-        ----------
-        worker : str
-            Address of worker to gather dependencies from
-        to_gather : list
-            Keys of dependencies to gather from worker -- this is not
-            necessarily equivalent to the full list of dependencies of ``dep``
-            as some dependencies may already be present on this worker.
-        total_nbytes : int
-            Total number of bytes for all the dependencies in to_gather combined
+        distributed.worker_state_machine.BaseWorker.gather_dep
         """
         if self.status not in WORKER_ANY_RUNNING:
             return None
 
         try:
-            self.log.append(("request-dep", worker, to_gather, stimulus_id, time()))
+            self.state.log.append(
```

Comment:
follow up: I don't think the worker class should log anythign on the state. I think these messages should be removed entirely, see also https://github.com/dask/distributed/pull/6442#issuecomment-1143860614